proxy/healthy_endpoints_test.go

```diff
@@ -173,20 +171,19 @@ func TestPHCForMultipleHealthyAndOneUnhealthyEndpoints(t *testing.T) {
 
 	for _, algorithm := range []string{"random", "consistentHash", "roundRobin", "powerOfRandomNChoices"} {
 		t.Run(algorithm, func(t *testing.T) {
-			tp, ps := setupProxy(t, fmt.Sprintf(`* -> backendTimeout("5ms") -> consistentHashKey("${request.header.ConsistentHashKey}") -> <%s, "%s", "%s", "%s">`,
+			mockMetrics, ps := setupProxy(t, fmt.Sprintf(`* -> backendTimeout("5ms") -> consistentHashKey("${request.header.ConsistentHashKey}") -> <%s, "%s", "%s", "%s">`,
 				algorithm, services[0].URL, services[1].URL, services[2].URL))
-			defer tp.close()
-			defer ps.Close()
 			failedReqs := sendGetRequests(t, ps)
+			mockMetrics.WithCounters(func(counters map[string]int64) {
+				assert.InDelta(t, float64(nRequests), float64(counters["passive-health-check.endpoints.dropped"]), 0.1*float64(nRequests)) // allow 10% error
```

Comment:
Shouldn't it be equal to `failedReqs`?
Also maybe move it after existing assert.InDelta

Reply:
> Shouldn't it be equal to `failedReqs`?

No because this metric will count number of requests drop across all endpoints, since we have 15000 request and one unhealthy endpoint then the counter should be equal 15000, however according to @RomanZavodskikh the code takes couple of requests before probability adjustment and dropping the endpoint hence it's less than 15000 and also the delta.